sound/soc/sof/ipc4-topology.c

```diff
@@ -126,6 +126,8 @@ static const struct sof_topology_token src_tokens[] = {
 static const struct sof_topology_token process_tokens[] = {
 	{SOF_TKN_PROCESS_PAYLOAD_WITH_OUTPUT_FMT, SND_SOC_TPLG_TUPLE_TYPE_BOOL, get_token_u16,
 		offsetof(struct sof_ipc4_process, payload_with_output_fmt)},
+	{SOF_TKN_PROCESS_IS_LOADABLE, SND_SOC_TPLG_TUPLE_TYPE_BOOL, get_token_u16,
+		offsetof(struct sof_ipc4_process, is_loadable)},
```

Comment:
Why do we need this token? If a module is not found in the base fw, shouldn't it be automatically loadable?

Reply:
we could add a flag to `struct sof_ipc4_fw_module` to signal that it is not in basefw:
In `sof_ipc4_load_library_by_uuid()` if the `lib_id != 0` then we set the flag for the modules in the library.

or we could move out the `SOF_IPC4_MOD_LIB_ID_SHIFT` to ipc4-priv.h, adding a mask or macro to check the lib_id, if it is 0 -> module is in basefw, if it is not 0 -> it is not in basefw (loaded as a library)

Reply:
```c
#define SOF_IPC4_MODULE_IS_IN_BASEFW(module)	\
	((((struct sof_ipc4_fw_module *)(module))->man4_module_entry.id & GENMASK(15, 12)) ? false : true)
```
```c
if (!SOF_IPC4_MODULE_IS_IN_BASEFW(&fw_lib->modules[i]))
```
```c
if (!SOF_IPC4_MODULE_IS_IN_BASEFW(widget->module_info))
```
but this is stretching the readability... A flag is better.

Reply:
but with a function (or inline function) it would be better:
```c
bool sof_ipc4_fw_module_is_in_basefw(struct sof_ipc4_fw_module *fw_module)
{
}
```

Reply:
I have added the flag and updated the PR.  every widget in its setup ops calls sof_ipc4_widget_setup_msg(), and in this function we setup swidget->module_info. we can access the flag simply by `swidget->module_info->ext_library`